client/src/api/v2/camera.ts

```diff
@@ -0,0 +1,294 @@
+  }, [device]);
+
+  const setActiveOverlay = useCallback(
+    (activeOverlay) => {
+      emit(
+        'set-overlays',
+        JSON.stringify({
+          [device]: activeOverlay,
+        }),
+      );
+    },
+    [device],
+  );
+
+  return { config, setActiveOverlay };
+}
+
+/**
+ * Sends a message to the rider's camera overlay.
+ *
+ * @param message Message to send to rider
+ */
+export function sendMessage(message: string) {
+  if (!message) return;
+
+  emit('send-message', message);
+}
+
+/**
+ * Send a message to server to start recording video
+ */
+export function startRecording() {
+  emit('start-camera-recording');
+}
+
+/**
+ * Send a message to server to stop recording video
+ */
+export function stopRecording() {
+  emit('stop-camera-recording');
+}
+
+const CameraRecordingOffPayload = Record({
+  status: Literal('off'),
+  /** Remaining disk space available for recording, in bytes */
+  diskSpaceRemaining: Number,
+});
+const CameraRecordingOnPayload = Record({
+  status: Literal('recording'),
+  /** Current duration of recording in minutes */
+  recordingMinutes: Number,
+  /** Path to video file being recorded */
+  recordingFile: String,
+  diskSpaceRemaining: Number,
+});
+const CameraRecordingErrorPayload = Record({
+  status: Literal('error'),
+  diskSpaceRemaining: Number,
+  /** Description of error that has occurred */
+  error: String,
+});
+const CameraRecordingStatusPayload = Union(
+  CameraRecordingOffPayload,
+  CameraRecordingOnPayload,
+  CameraRecordingErrorPayload,
+);
+type CameraRecordingStatusPayload = Static<typeof CameraRecordingStatusPayload>;
+
+interface CameraRecordingStatusItem {
+  /** Display name of the status item e.g. "Disk Space Remaining" */
+  name: string;
+  /** Value of the status item e.g. "1.2 GiB" */
+  value: string;
+}
+
+/**
+ * Parse the recording payload into an object and format the values
+ *
+ * Payload structure is defined in the 'V3 MQTT Topics' page on Notion.
+ * Topic is /v3/camera/recording/status/<primary/>secondary>.
+ *
+ * @param payload Payload from MQTT message, parsed
+ * @returns Formatted payload
+ */
+export function formatRecordingPayload(payload?: CameraRecordingStatusPayload) {
+  if (!payload) return null;
+
+  // Format data always present regardless of status
+  const commonData: CameraRecordingStatusItem[] = [
+    { name: 'Status', value: capitalise(payload.status) },
+    {
+      name: 'Disk space remaining',
+      value: formatBytes(payload.diskSpaceRemaining),
+    },
+  ];
+
+  // Format data that is dependent on status
+  const formatter = CameraRecordingStatusPayload.match(
+    () => [],
+    (onPayload) => [
+      {
+        name: 'Recording duration',
+        value: formatMinutes(onPayload.recordingMinutes),
+      },
+      { name: 'Recording file', value: onPayload.recordingFile },
+    ],
+    (errorPayload) => [{ name: 'Error', value: errorPayload.error }],
+  );
+
+  return [...commonData, ...formatter(payload)];
```

Comment:
I think the use of `concat` here instead of arrays destructuring into an array would be more understandable, but this looks neater.